src/ui/BlendEngine.js

```diff
@@ -142,8 +146,19 @@ exports = Class(View, function () {
     this._canvas = new Canvas({ width: MAX_TEX_WIDTH, height: MAX_TEX_HEIGHT, useWebGL: true });
   };
 
-  this.obtainParticleArray = function (count) {
-    for (var i = 0; i < count; i++) {
+  this.obtainParticleArray = function (count, opts) {
+    var isBrowser = (userAgent.APP_RUNTIME === 'browser');
+    var isMobile = (userAgent.DEVICE_TYPE === 'mobile');
+    var isSimulator = userAgent.SIMULATED;
+   
+    opts = opts || {};
+
+    // TODO: disable blend engine on mobile browsers until the performance is improved
+    var currCount = (isBrowser && isMobile && !isSimulator)
+      ? 0
+      : performance.getAdjustedParticleCount(count, opts.effectPerformanceScore, opts.allowReduction);
```

Comment:
Please change all instances of `effectPerformanceScore` to just `performanceScore`